libraries/version/src/version_impl.cpp.in

```diff
@@ -0,0 +1,36 @@
+/**
+ *  @file version_impl.cpp.in
+ *  @copyright defined in eos/LICENSE
+ *  \warning This file is machine generated. DO NOT EDIT.  See version_impl.cpp.in for changes.
+ */
+
+#include "version_impl.hpp"
+
+namespace eosio { namespace version {
+
+   const std::string version_major {"@_VERSION_MAJOR_@" };
+   const std::string version_minor {"@_VERSION_MINOR_@" };
+   const std::string version_patch {"@_VERSION_PATCH_@" };
+   const std::string version_suffix{"@_VERSION_SUFFIX_@"};
+   const std::string version_hash  {"@_VERSION_HASH_@"  };
+   const bool        version_dirty { @_VERSION_DIRTY_@  };
+
+   const std::string& _version_client() {
+      static const std::string version{'v' + version_major +
+                                       '.' + version_minor +
+                                       '.' + version_patch +
+                                       '-' + version_suffix};
+      return version;
+   }
+
+   const std::string& _version_full() {
```

Comment:
This is going to potentially give a different version than what nodeos reports (and if we modify nodeos to use the new version string it means we're going to lose some information). Is that intentional?

Reply:
Yes, the plan was to have this specific versioning output from now on.
This functionality is also planned to be added into `nodeos` and the other programs alike as well if everything is agreed upon.